swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/work/WorkQueueManager.kt

```diff
@@ -84,7 +86,16 @@ class WorkQueueManager(
     }
   }
 
-  private fun isEnabled(): Boolean {
+  /**
+   * Swabbie is considered disabled if it is out of service in discovery, disabled via fast property, or the
+   *  schedule determines it is off hours.
+   */
+  internal fun isEnabled(): Boolean {
+    if (!up.get()) {
```

Comment:
This also means disabling one instance in discovery stops healthy and UP instances due to the else condition in `monitor`. Another approach that avoids this sort of problem would be to only gate `WorkProcessor.process()` on discovery state.

Reply:
Great catch. Thanks. I'm going to make the work processor also discovery activated so that it only processes when it's up.